docs/copied-from-beats/outputconfig.asciidoc

```diff
@@ -340,6 +362,70 @@ This configuration results in indices named `sev1`, `sev2`, and `sev3`.
 
 The `mappings` setting simplifies the configuration, but is limited to string
 values. You cannot specify format strings within the mapping pairs.
+endif::apm-server[]
+
+ifdef::apm-server[]
+The following example sets the index based on whether the `processor.event` field
+contains the specified string:
+
+["source","yaml",subs="attributes"]
+------------------------------------------------------------------------------
+output.elasticsearch:
+  hosts: ["http://localhost:9200"]
+  indices:
+   - index: "apm-%{[observer.version]}-sourcemap"
+      when.contains:
+        processor.event: "sourcemap"
+  
+   - index: "apm-%{[observer.version]}-error-%{+yyyy.MM.dd}"
+      when.contains:
+        processor.event: "error"
+  
+   - index: "apm-%{[observer.version]}-transaction-%{+yyyy.MM.dd}"
+      when.contains:
+        processor.event: "transaction"
+  
+   - index: "apm-%{[observer.version]}-span-%{+yyyy.MM.dd}"
+      when.contains:
+        processor.event: "span"
+  
+   - index: "apm-%{[observer.version]}-metric-%{+yyyy.MM.dd}"
+      when.contains:
+        processor.event: "metric"
+  
+   - index: "apm-%{[observer.version]}-onboarding-%{+yyyy.MM.dd}"
+      when.contains:
+        processor.event: "onboarding"
```

Comment:
I was planning on simplifying the default config a bit (within the ILM PR). As you described in this PR it is now also possible to use `processor.event` in the regex, leading to `index: "apm-%{[observer.version]}-%{[processor.event]}-%{+yyyy.MM.dd}"`.  With this we can simplify the handling without any changes in the outcome. 
```
output.elasticsearch:
  hosts: ["http://localhost:9200"]
  index: "apm-%{[observer.version]}-%{[processor.event]}-%{+yyyy.MM.dd}"
  indices:
   - index: "apm-%{[observer.version]}-sourcemap"
      when.contains:
        processor.event: "sourcemap"
```

But as this change is _only in my head_ so far, what you have looks great!

Reply:
Woah, does it work in that order? I thought APM Server went with the first match. I was playing around with basically the same idea yesterday and got it to work in this order:
```
output.elasticsearch:
  hosts: ["http://localhost:9200"]
  indices:
   - index: "apm-%{[observer.version]}-sourcemap"
      when.contains:
        processor.event: "sourcemap"
   - index: "apm-%{[observer.version]}-%{[processor.event]}-%{+yyyy.MM.dd}"
```

Reply:
If you're planning on simplifying the default config within the ILM PR, let's definitely get on the same page. I'd love to update this documentation with what you're thinking.